openmock.go

```diff
@@ -5,36 +5,56 @@ import (
-	KafkaEnabled          bool     `env:"OPENMOCK_KAFKA_ENABLED" envDefault:"false"`
-	KafkaClientID         string   `env:"OPENMOCK_KAFKA_CLIENT_ID" envDefault:"openmock"`
-	KafkaSeedBrokers      []string `env:"OPENMOCK_KAFKA_SEED_BROKERS" envDefault:"kafka:9092" envSeparator:","`
-	AMQPEnabled           bool     `env:"OPENMOCK_AMQP_ENABLED" envDefault:"false"`
-	AMQPURL               string   `env:"OPENMOCK_AMQP_URL" envDefault:"amqp://guest:guest@rabbitmq:5672"`
-	RedisType             string   `env:"OPENMOCK_REDIS_TYPE" envDefault:"memory"`
-	RedisURL              string   `env:"OPENMOCK_REDIS_URL" envDefault:"redis://redis:6379"`
-	CorsEnabled           bool     `env:"OPENMOCK_CORS_ENABLED" envDefault:"false"`
-
-	// Customized pipeline functions
+	LogLevel              string `env:"OPENMOCK_LOG_LEVEL" envDefault:"info"`
+	TemplatesDir          string `env:"OPENMOCK_TEMPLATES_DIR" envDefault:"./templates"`
+	TemplatesDirHotReload bool   `env:"OPENMOCK_TEMPLATES_DIR_HOT_RELOAD" envDefault:"true"`
+
+	// HTTP channel
+	HTTPEnabled bool   `env:"OPENMOCK_HTTP_ENABLED" envDefault:"true"`
+	HTTPPort    int    `env:"OPENMOCK_HTTP_PORT" envDefault:"9999"`
+	HTTPHost    string `env:"OPENMOCK_HTTP_HOST" envDefault:"0.0.0.0"`
+	CorsEnabled bool   `env:"OPENMOCK_CORS_ENABLED" envDefault:"false"`
+
+	// Admin channel
+	AdminHTTPEnabled bool   `env:"OPENMOCK_ADMIN_HTTP_ENABLED" envDefault:"true"`
+	AdminHTTPPort    int    `env:"OPENMOCK_ADMIN_HTTP_PORT" envDefault:"9998"`
+	AdminHTTPHost    string `env:"OPENMOCK_ADMIN_HTTP_HOST" envDefault:"0.0.0.0"`
+
+	// Kafka channel
+	KafkaEnabled     bool     `env:"OPENMOCK_KAFKA_ENABLED" envDefault:"false"`
+	KafkaClientID    string   `env:"OPENMOCK_KAFKA_CLIENT_ID" envDefault:"openmock"`
+	KafkaSeedBrokers []string `env:"OPENMOCK_KAFKA_SEED_BROKERS" envDefault:"kafka:9092" envSeparator:","`
+
+	// AMQP channel
+	AMQPEnabled bool   `env:"OPENMOCK_AMQP_ENABLED" envDefault:"false"`
+	AMQPURL     string `env:"OPENMOCK_AMQP_URL" envDefault:"amqp://guest:guest@rabbitmq:5672"`
+
+	// Redis configuration for admin's ephemeral storage or redis commands
+	RedisType string `env:"OPENMOCK_REDIS_TYPE" envDefault:"memory"`
+	RedisURL  string `env:"OPENMOCK_REDIS_URL" envDefault:"redis://redis:6379"`
+
+	// GRPC channel
+	GRPCEnabled bool   `env:"OPENMOCK_GRPC_ENABLED" envDefault:"false"`
+	GRPCPort    int    `env:"OPENMOCK_GRPC_PORT" envDefault:"50051"`
+	GRPCHost    string `env:"OPENMOCK_GRPC_HOST" envDefault:"0.0.0.0"`
+
+	//////////////////////// Customized functions //////////////////////////////////
+	// KafkaConsumePipelineFunc is a pipeline function run to when consume a message
 	KafkaConsumePipelineFunc KafkaPipelineFunc
+	// KafkaPublishPipelineFunc is a pipeline function run to when produce a message
 	KafkaPublishPipelineFunc KafkaPipelineFunc
+	// GRPCServiceMap is a map of gRPC [service_name => GRPCService]
+	GRPCServiceMap map[string]GRPCService
```

Comment:
instead of saving it as package global, moved to instance field for better encapsulation.